gradio/components.py

```diff
@@ -237,23 +237,28 @@ def hash_url(self, url: str, chunk_num_blocks: int = 128) -> str:
             sha1.update(data)
         return sha1.hexdigest()
 
-    def hash_base64(self, base64_encoding: str, chunk_num_blocks: int = 128) -> str:
+    def _hash(self, base64_encoding: str | bytes, chunk_num_blocks: int = 128):
         sha1 = hashlib.sha1()
         for i in range(0, len(base64_encoding), chunk_num_blocks * sha1.block_size):
             data = base64_encoding[i : i + chunk_num_blocks * sha1.block_size]
-            sha1.update(data.encode("utf-8"))
+            sha1.update(data.encode("utf-8") if isinstance(data, str) else data)
         return sha1.hexdigest()
```

Comment:
You'd save cycles by letting the hasher do the work if you don't need to encode the data.
```suggestion
        if isinstance(base64_encoding, bytes):
            sha1.update(base64_encoding)
        else:
            for i in range(0, len(base64_encoding), chunk_num_blocks * sha1.block_size):
                data = base64_encoding[i : i + chunk_num_blocks * sha1.block_size]
                sha1.update(data.encode("utf-8"))
        return sha1.hexdigest()

Reply:
Thanks for the suggestion!

Reply:
Thanks for the suggestion. Done!